testnet/stacks-node/src/config.rs

```diff
@@ -671,7 +775,39 @@ impl Config {
     }
 
     pub fn from_config_file(config_file: ConfigFile) -> Result<Config, String> {
-        let default_node_config = NodeConfig::default();
+        if config_file.burnchain.as_ref().map(|b| b.mode.clone()) == Some(Some("mockamoto".into()))
+        {
+            let default = Self::from_config_default(ConfigFile::mockamoto(), None)?;
+            Self::from_config_default(config_file, Some(default))
+        } else {
+            Self::from_config_default(config_file, None)
+        }
+    }
+
+    fn from_config_default(
```

Comment:
This is just a personal preference. But I would make from_config_default always accept a Config (nonoptional) and then call it with Config::default() at line 781.

Reply:
Yep, agreed. Updated in b447984